### PR TITLE
[SPARK-45299][TESTS] Remove JDK 8 workaround in UtilsSuite

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR removes the legacy workaround for JDK 7 and below at SPARK-12486. The main code was cleaned up at SPARK-16182 but the test code was not cleaned up.

### Why are the changes needed?

To remove legacy workaround.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Fixed unittests.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @srowen

File: core/src/test/scala/org/apache/spark/util/UtilsSuite.scala [hunk 1]
```diff
@@ -1018,37 +1018,35 @@ class UtilsSuite extends SparkFunSuite with ResetSystemProperties {
...
+           |#!/usr/bin/env bash
+           |trap "" SIGTERM
+           |sleep 10
+         """.stripMargin
+      Files.write(cmd.getBytes(UTF_8), file)
+      file.getAbsoluteFile.setExecutable(true)
+
+      val process = new ProcessBuilder(file.getAbsolutePath).start()
+      val pid = getPid(process)
+      assert(pidExists(pid))
```
Comment (yaooqinn) [inline core/src/test/scala/org/apache/spark/util/UtilsSuite.scala:1037]:
```suggestion
      process.toHandle.pid()
```

**Review by yaooqinn — APPROVED** [review]

File: core/src/test/scala/org/apache/spark/util/UtilsSuite.scala [hunk 2]
```diff
@@ -1018,37 +1012,34 @@ class UtilsSuite extends SparkFunSuite with ResetSystemProperties {
...
+           |trap "" SIGTERM
+           |sleep 10
+         """.stripMargin
+      Files.write(cmd.getBytes(UTF_8), file)
+      file.getAbsoluteFile.setExecutable(true)
+
+      val process = new ProcessBuilder(file.getAbsolutePath).start()
+      process.toHandle.pid()
+      try {
+        signal(pid, "SIGSTOP")
```
Comment (yaooqinn) [inline core/src/test/scala/org/apache/spark/util/UtilsSuite.scala:1032]: The scope of pid here is changed, we need to change it to var and re-assign

**Review by srowen — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
This seems to have several compilation issues and one import issue, @HyukjinKwon . Could you double-check?

```
$ build/sbt "core/testOnly *.UtilsSuite"
...
[error] /Users/dongjoon/APACHE/spark-merge/core/src/test/scala/org/apache/spark/util/UtilsSuite.scala:1003:26: type mismatch;
[error]  found   : Long
[error]  required: Int
[error]         assert(pidExists(pid))
[error]                          ^
[error] /Users/dongjoon/APACHE/spark-merge/core/src/test/scala/org/apache/spark/util/UtilsSuite.scala:1009:27: type mismatch;
[error]  found   : Long
[error]  required: Int
[error]         assert(!pidExists(pid))
[error]                           ^
[error] /Users/dongjoon/APACHE/spark-merge/core/src/test/scala/org/apache/spark/util/UtilsSuite.scala:1012:16: type mismatch;
[error]  found   : Long
[error]  required: Int
[error]         signal(pid, "SIGKILL")
[error]                ^
[info] Long <: Int?
[info] false
[error] /Users/dongjoon/APACHE/spark-merge/core/src/test/scala/org/apache/spark/util/UtilsSuite.scala:1032:16: type mismatch;
[error]  found   : Long
[error]  required: Int
[error]         signal(pid, "SIGSTOP")
[error]                ^
[info] Long <: Int?
[info] false
[error] /Users/dongjoon/APACHE/spark-merge/core/src/test/scala/org/apache/spark/util/UtilsSuite.scala:1040:27: type mismatch;
[error]  found   : Long
[error]  required: Int
[error]         assert(!pidExists(pid))
[error]                           ^
[error] /Users/dongjoon/APACHE/spark-merge/core/src/test/scala/org/apache/spark/util/UtilsSuite.scala:1042:16: type mismatch;
[error]  found   : Long
[error]  required: Int
[error]         signal(pid, "SIGKILL")
[error]                ^
[info] Long <: Int?
[info] false
[error] /Users/dongjoon/APACHE/spark-merge/core/src/test/scala/org/apache/spark/util/UtilsSuite.scala:35:34: Unused import
[error] import org.apache.commons.lang3.{JavaVersion, SystemUtils}
[error]                                  ^
[warn] one warning found
[error] 7 errors found
[error] (core / Test / compileIncremental) Compilation failed
[error] Total time: 6 s, completed Sep 25, 2023, 10:11:36 AM
```

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you.

I verified manually too. Merged to master.

```
[info] UtilsSuite:
[info] - timeConversion (19 milliseconds)
[info] - Test byteString conversion (4 milliseconds)
[info] - bytesToString (3 milliseconds)
[info] - copyStream (6 milliseconds)
[info] - copyStreamUpTo (23 milliseconds)
[info] - memoryStringToMb (1 millisecond)
[info] - splitCommandString (2 milliseconds)
[info] - string formatting of time durations (1 millisecond)
[info] - reading offset bytes of a file (51 milliseconds)
[info] - reading offset bytes of a file (compressed) (17 milliseconds)
[info] - reading offset bytes across multiple files (9 milliseconds)
[info] - reading offset bytes across multiple files (compressed) (7 milliseconds)
[info] - deserialize long value (0 milliseconds)
[info] - writeByteBuffer should not change ByteBuffer position (1 millisecond)
[info] - get iterator size (1 millisecond)
[info] - getIteratorZipWithIndex (2 milliseconds)
[info] - SPARK-35907: createDirectory (16 milliseconds)
[info] - doesDirectoryContainFilesNewerThan (7 milliseconds)
[info] - resolveURI (1 millisecond)
[info] - resolveURIs with multiple paths (2 milliseconds)
[info] - nonLocalPaths (1 millisecond)
[info] - isBindCollision (3 milliseconds)
[info] - log4j log level change (1 millisecond)
[info] - deleteRecursively (17 milliseconds)
[info] - loading properties from file (11 milliseconds)
[info] - timeIt with prepare (2 seconds, 12 milliseconds)
[info] - fetch hcfs dir (153 milliseconds)
[info] - shutdown hook manager (2 milliseconds)
[info] - isInDirectory (2 milliseconds)
[info] - circular buffer: if nothing was written to the buffer, display nothing (1 millisecond)
[info] - circular buffer: if the buffer isn't full, print only the contents written (0 milliseconds)
[info] - circular buffer: data written == size of the buffer (0 milliseconds)
[info] - circular buffer: multiple overflow (0 milliseconds)
[info] - isDynamicAllocationEnabled (3 milliseconds)
[info] - getDynamicAllocationInitialExecutors (5 milliseconds)
[info] - Set Spark CallerContext (11 milliseconds)
[info] - encodeFileNameToURIRawPath (0 milliseconds)
[info] - decodeFileNameInURI (0 milliseconds)
[info] - Kill process (33 milliseconds)
[info] - chi square test of randomizeInPlace (15 milliseconds)
[info] - redact sensitive information (2 milliseconds)
[info] - redact sensitive information in command line args (2 milliseconds)
[info] - redact sensitive information in sequence of key value pairs (1 millisecond)
[info] - tryWithSafeFinally (2 milliseconds)
[info] - tryWithSafeFinallyAndFailureCallbacks (7 milliseconds)
[info] - load extensions (3 milliseconds)
[info] - check Kubernetes master URL (1 millisecond)
[info] - stringHalfWidth (0 milliseconds)
[info] - trimExceptCRLF standalone (2 milliseconds)
[info] - pathsToMetadata (1 millisecond)
[info] - checkHost supports both IPV4 and IPV6 (1 millisecond)
[info] - checkHostPort support IPV6 and IPV4 (0 milliseconds)
[info] - parseHostPort support IPV6 and IPV4 (0 milliseconds)
[info] - executorOffHeapMemorySizeAsMb when MEMORY_OFFHEAP_ENABLED is false (2 milliseconds)
[info] - executorOffHeapMemorySizeAsMb when MEMORY_OFFHEAP_ENABLED is true (1 millisecond)
[info] - executorMemoryOverhead when MEMORY_OFFHEAP_ENABLED is true, but MEMORY_OFFHEAP_SIZE not config scene (1 millisecond)
[info] - isPushBasedShuffleEnabled when PUSH_BASED_SHUFFLE_ENABLED and SHUFFLE_SERVICE_ENABLED are both set to true in YARN mode with maxAttempts set to 1 (78 milliseconds)
[info] Run completed in 3 seconds, 332 milliseconds.
[info] Total number of tests run: 57
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 57, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 22 s, completed Sep 25, 2023, 8:16:56 PM
```

**Pull request closed without merging**